Issue 3788 - subcategories for copy past

moroz-art-dev opened the pull request:
# Description
The idea is to allow not to transfer all styles when copying, but to select the ones you need

Fixes #3788

# How Has This Been Tested?

it is necessary to test different options, since the function is new, perhaps something can be improved
It is necessary to see how the mechanism works in different cases

https://user-images.githubusercontent.com/88348091/199775378-37f208de-741d-4401-8b1a-25f2e0d54f46.mov



# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

Olekrut reviewed (changes requested):
Previous comment and.. 
6) Need more deep test for all options!!

elzadj committed:
Add google map  errorscatching

elzadj committed:
Add fadeIn for the map error message

elzadj committed:
remove a wrong config file

elzadj committed:
Clean code

Rustam198804 committed:
Fix code spaces

myroslavve committed:
Issue 3909 - reusable blocks with maxi blocks (#4304)

* fix reusable blocks wrong `uniqueID` generation

* create method to observe width of a clone of a block that is on top level
update width of a block if a width of a clone changes.

* fix test

* fix after merge

* fix block styles disappearing after block preview

Fybex committed:
Issue 4152 - `TransformControl` hover values inherit from normal fix (#4404)

* `SquareControl`: re-render on bp change

fix `TransformControl` responsive `onChange` with handlers(`BlockResizer`)

* `TransformControl`: fix in hover/normal inherit

Marko64 committed:
Create mvp.yml (#4409)

elzadj committed:
Merge pull request #4410 from maxi-blocks/map-message-styles

Map no key message styles

elzadj committed:
Fix conflicts

Marko64 committed:
Update editor.scss

Marko64 committed:
Update editor.scss

elzadj committed:
Merge pull request #4182 from maxi-blocks/issue-4175-text-border-affecting-width

Issue 4175 - Text border affecting width

myroslavve committed:
fix console error when importing SC

myroslavve committed:
fix import/export test timings

myroslavve committed:
abstract `copySCToEdit`

myroslavve committed:
copy style cards before editing

myroslavve committed:
fix global test trying to save already imported template

Fybex committed:
Issue 3487 - Transition transform support (#4390)

* Fix imports order

* Create `getTransformTransition` helper

* `transitionAttributesCreator`: transform attrs

 - transform params to object
- add selectors param to add transform settings to transition schema

* `inspector-transition`: clean transform keys

when no hover

* `getTransitionStyles`: `isTransform` exception

* Split ordered and unordered selectors

* `getTransformTransition`: use transform selectors

* `getTransitionData`: return transform data as well

* `inlineStyles `: avoid console error

* `inspector-transition`: selectors param

* `inspector-transition`: without canvas tab labels

* Add transform transition unit tests

* Remove unused variables/imports

* Explanation comment

* Add transform bg layers transition support

* `styleProcessor`: fx selectors if !data.transition

* Don't add transition attrs for hover layers

* Update text-maxi e2e

custom css targets was changed

* Fix settings target display

* Support `transform-origin` transition

Marko64 committed:
Update editor.scss

elzadj committed:
add languages folder

Marko64 committed:
Serial removed for patterns

elzadj committed:
Add s special message for empty API key

elzadj committed:
clean code

Marko64 committed:
Update MasonryItem.js

Marko64 committed:
Update editor.scss

elzadj committed:
generate .pot file

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into translations-pot-file

elzadj committed:
Merge pull request #4425 from maxi-blocks/translations-pot-file

Translations pot file

Marko64 committed:
WrapClassName

Marko64 committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into column-outline-after-resetting

elzadj committed:
Merge pull request #4407 from maxi-blocks/issue-4406-modal-titles

Issue 4406 - Modal items titles

myroslavve committed:
fix import/export test

myroslavve committed:
fix svg markers size not reseting on reset

Marko64 committed:
Baby toolbar z-index

Marko64 committed:
Issue 4393 - Block indicators z-index (#4413)

Fybex committed:
Issue 4124 - Column value on responsive resetting (#4420)

* `getColumnDefaultValue`: reset to und if `!general`

* column-maxi e2e: wait for selectors

elzadj committed:
Merge pull request #4421 from maxi-blocks/issue-4349-SC-tests

Issue 4349 - Style cards tests

Marko64 committed:
Issue 4419 - Show/hide

Marko64 committed:
Update editor.scss

Marko64 committed:
bem

elzadj committed:
Merge pull request #4414 from maxi-blocks/issue-4411-copy-paste-UI

Issue 4411 - Copy/paste UI

elzadj committed:
Merge pull request #4426 from maxi-blocks/issue-4416-baby-toolbar-overlapping-toolbar-dropdown

Issue 4416 - Baby toolbar z-index

myroslavve committed:
fix markers indent on size change

myroslavve committed:
fix indent for custom markers

Marko64 committed:
Update editor.scss

Naaaaiix committed:
Fix old broken transitions

Fybex committed:
Issue 4388 - `svg-icon-maxi` fit-height support (#4422)

* Merge `getIconSize` and `getSVGStyles`

* `svg-icon-maxi`: fit-height support

* `togglePreserveAspectRatio`

* Use `togglePreserveAspectRatio`

* Update e2e

height property was added

Marko64 committed:
Issue 4392 - Scroll effects default values (#4401)

* Update shortcuts.js

* Rotate fixed

* Fade

* Blur

* Labels

* Update handleSetAttributes.js

* Values, styles

* Values

* Values final final 3

* order

Naaaaiix committed:
Fixed default transitions for `transitionMigrator`

Naaaaiix committed:
Fixes issue #4437

elzadj committed:
Merge pull request #4412 from maxi-blocks/map-fixes

Map fixes

elzadj committed:
Merge pull request #4439 from maxi-blocks/issue-4437

Issue 4437 - Margin conflict on full width Containers

elzadj committed:
Merge pull request #4438 from maxi-blocks/transition-migrator

Transition migrator from old broken patterns

elzadj committed:
Merge pull request #4428 from maxi-blocks/issue-4419-show-hide

Issue 4419 - Show/hide fixes

elzadj committed:
Merge pull request #4250 from maxi-blocks/column-outline-after-resetting

Fix column outline after resetting

Naaaaiix committed:
v1.0.0-RC1 (#4423)

* Update to v1.0.0-RC1

* Fixes issue #4437

* Merge pull request #4439 from maxi-blocks/issue-4437

Issue 4437 - Margin conflict on full width Containers

* Merge pull request #4438 from maxi-blocks/transition-migrator

Transition migrator from old broken patterns

---------

Co-authored-by: Svitlana <2401618+elzadj@users.noreply.github.com>

Naaaaiix committed:
Fixes #4447

myroslavve committed:
fix IB relations dissapearing on switch to code editor (#4453)

Fybex committed:
Issue 4368 - Replace bg shape image in copypasted block fix (#4405)

* Change the svg data id if different uniqueID

* `handleBGLayersOnUniqueIDChange`

* Remove old idea, implement new

call `handleBGLayersOnUniqueIDChange` on duplication/copy paste

* Fix `handleBGLayersOnUniqueIDChange` calls

* `handleBGLayersOnUniqueIDChange`: return if no data

* `handleAttributesOnPaste`: 2 ifs into one

* Fix inherit problems, adjust code to structure

By structure I mean `propagateNewUniqueID`

* `propagateNewUniqueID`: move `if` to right place

* `getUpdatedBGLayersWithNewUniqueID`: optimization

* Don't use `blockAttributesUpdate` for bgLayers

* Extract old uniqueID correctly

Naaaaiix committed:
Fixes multiple font loads on frontend for SC

Marko64 committed:
Issue 4411 - Special paste UI (#4442)

* Update editor.scss

* Removed bg

elzadj committed:
Merge pull request #4452 from maxi-blocks/issue-4447

Issue 4447 - Only the last font weight from the page is loaded on frontend

Naaaaiix committed:
Avoid CustomCSS errors

myroslavve committed:
update snapshots after fix

myroslavve committed:
`createTextWithList` function

myroslavve committed:
add test for list with svg marker

myroslavve committed:
add test for options with different units

myroslavve committed:
add rtl test for list indent options

dependabot[bot] committed:
Bump dns-packet from 5.3.1 to 5.4.0 (#4474)

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 5.3.1 to 5.4.0.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v5.3.1...5.4.0)

---
updated-dependencies:
- dependency-name: dns-packet
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

elzadj committed:
Merge pull request #4472 from maxi-blocks/test-validate-css

Fix CustomCSS validations fake positives in tests

elzadj committed:
Update typesense node url

Naaaaiix committed:
Lazy load and Suspense for blocks and inspectors (#4443)

* Change the svg data id if different uniqueID

* `handleBGLayersOnUniqueIDChange`

* Remove old idea, implement new

call `handleBGLayersOnUniqueIDChange` on duplication/copy paste

* Fix `handleBGLayersOnUniqueIDChange` calls

* `handleBGLayersOnUniqueIDChange`: return if no data

* `handleAttributesOnPaste`: 2 ifs into one

* Fix inherit problems, adjust code to structure

By structure I mean `propagateNewUniqueID`

* `propagateNewUniqueID`: move `if` to right place

* `getUpdatedBGLayersWithNewUniqueID`: optimization

* Don't use `blockAttributesUpdate` for bgLayers

* Implement lazy and Suspense for blocks
Implement Suspense for Inspector

* Fix maxiBlocks store reducer

* Extract old uniqueID correctly

* Fixed Suspense for pasted blocks
Fixed Suspense when swapping from code editor
Improved speed of Suspense

* Update tests

* Fix tests

* Implement PuffLoader for Suspense blocks

* Simplified code

* Replace Spinner for ContentLoader

* Right colour and size for loader. This worked best of all of them.

<PuffLoader   color="#ff4a17"   size={20}   speedMultiplier={0.8} />

* Implement e2e util `insertMaxiBlock` for Suspense

* Fix some tests

* Fix Icon Maxi tests

---------

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>
Co-authored-by: Mister_C <7437580+ckp267@users.noreply.github.com>

elzadj committed:
Merge pull request #4477 from maxi-blocks/typesense-node-update

Update typesense node url

Marko64 committed:
Update index.js (#4481)

myroslavve committed:
fix image disappearing on changing shape mask

myroslavve committed:
fix for case where layer order is not layer index

Marko64 committed:
Update index.js (#4493)

elzadj committed:
Fix .interface-interface-skeleton__content bug

elzadj committed:
Change code from render to createRoot

elzadj committed:
Fix returns in useEffect

elzadj committed:
small fixes

elzadj committed:
Fix another userEffect returning null

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into fix-wp-62-beta4-bugs

elzadj committed:
Fix bug with resizeObserverSelector

elzadj committed:
Fix the resizeObserver class

elzadj committed:
remove log

elzadj committed:
Fix responsive switcher breaking the editor

myroslavve committed:
fix shape ratio and position dropping on image replace

myroslavve committed:
fix for replacing SVGs

Fybex committed:
`getSVGWidthStyles`: `disableHeight` prop

true by default, false for `getIconSize`

Fybex committed:
Update/add tests

myroslavve committed:
Merge branch 'master' into issue-4359-list-markers

Fybex committed:
Explanation comment

myroslavve committed:
fix list style none indent

Fybex committed:
Copy paste: exclude icon content

from:
- search-maxi
- map-maxi
- button-maxi
- accordion-maxi
- slider-maxi

Fybex committed:
`accordion-maxi`: add ability to copy active icon

myroslavve committed:
fix for list style custom size

Fybex committed:
Update map-maxi e2e

remove height

elzadj committed:
Merge pull request #4484 from maxi-blocks/issue-4464-shape-mask-image

Issue 4464 - fix image disappearing on changing shape mask

elzadj committed:
Merge pull request #4497 from maxi-blocks/issue-4451

Issue 4451 - Don't set unnecessary height for svg-icon-maxi

elzadj committed:
Merge pull request #4498 from maxi-blocks/issue-4486

Issue 4486 - content is pasted on 'Paste styles - all' part 2

elzadj committed:
Fix broken maxi madal

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into fix-wp-62-beta4-bugs

elzadj committed:
Add backward compatibility for WP 6.1.1 and below (react below 18)

elzadj committed:
Fix shape mask

elzadj committed:
Merge pull request #4473 from maxi-blocks/issue-4359-list-markers

Issue 4359 - list markers

elzadj committed:
Fix svg upload button

elzadj committed:
Fix broken GradientPicker

elzadj committed:
Fix FSE breaking on last Maxi block removal

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into fix-wp-62-beta4-bugs

elzadj committed:
Add maxi button back to FSE toolbar

elzadj committed:
Fix z-index for responsive toolbar in FSE

elzadj committed:
Remove logs

elzadj committed:
Fix maxi button position in FSE

elzadj committed:
Remove logs

Marko64 committed:
Update button-maxi.js

elzadj committed:
Add preview images to Image and Button blocks

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into Blocks-preview

elzadj committed:
Remove Fragments

Marko64 committed:
Issue 4499 - Paste part 3 (#4501)

* Video image

* Update handleSetAttributes.js

* Update data.js

Fybex committed:
`image-maxi`: canvas max-height fix (#4503)

Fybex committed:
Issue 4360 - Correct column size styles on small devices (#4506)

* `getColumnSizeStyles`: find column size correctly

* Update e2e

Marko64 committed:
Update editor.scss (#4502)

Naaaaiix committed:
Issues 4487 & 4489 - Suspense Lazy load issues (#4491)

* Update LazySuspense load to fix #4487

* Fix withSuspense lifecycle

* Fix tests

* Fix tests

* Fix test

* Improve withSuspense lifecycle

* Fix tests

* Fixes issues with column width
Minor code improvements and comments

* Added `allBlocksHaveBeenRendered` selector

Naaaaiix committed:
Issue 3388 - DC (#4430)

* Issue 3388 - master copy add files

* Issue 3388 - master copy, fix

* Issue 3388 - fix bag previev

* Issue 3388 - fix bag previev and add select params date and limit

* Issue 3388 - dev date

* Issue 3388 - dev date

* Issue 3388 - dev date and a bit of decomposition DC

* Issue 3388 - add component date and add date customization settings

* Issue 3388 - code decomposition

* Issue 3388 - clean console.log

* Issue 3388 - new date input

* Issue 3388 - update test

* Issue 3388 - date popup format

* Issue 3388 - limit: clear html + limit max, update input date

* Issue 3388 - add short Month and short day in inpute date

* Issue 3388 - fix css

* Update src/components/toolbar/components/date-formatting/index.js

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

* Update src/components/toolbar/components/date-formatting/index.js

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

* Update src/components/toolbar/components/date-formatting/index.js

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

* Update src/components/toolbar/components/date-formatting/index.js

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

* Update src/blocks/text-maxi/save.js

Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

* Issue 3388 - fix edits Fybex

* Issue 3388 - delete comment eslint

* Issue 3388 - fix issue eslint

* Issue 3388 - update hooks eslint

* Issue 3388 - update date f eslint

* Issue 3388 - fix author

* Issue 3388 - fix id + author

* Issue 3388 - fixed some edits

* Issue 3388 - fix categorizes

* Issue 3388 - code date update

* Issue 3388 - dev

* Issue 3388 - fix link

* Issue 3388 - fix link 2

* Issue 3388 - fix author

* Issue 3388 - fix await and fix type number

* Issue 3388 - fix visible select on error

* Issue 3388 - fixed translation

* Fix translation and labels for DateFormat

* Add class to popover

* Fix author by default

* Fix labels and translations

* simplify the code

* Fix button line-height in SC

* server rendering

* More fixes

* Add api requests for frontend

* Add rest API url

* Adjust code to work with custom meta

* process posts, pages, site

* process media, authors, categories and tags

* Small fixes

* Fix Site settings craching the block

* Fix author name

* Fix date format on frontend

* Add endpoint for site settings

* Add endpoint for author

* Fix bug with Site type dwropdown on first load

* Fix Custom date

* Fix small bugs, format code

* Fix rendering on frontend

* remove logs

* Fix button attributes

* Fix cutom time zones on backend

* Fix cistom time zone for frontend

* Add excerpt limit to frontend

* Move limit funtrions to utils

* Fix invalid date on frontent

* Fix small code issues

* Fix package-lock.json

* Fix package-lock.json again

* Fix api requests for the rest api url with a parameter on backend

* Fix first separator for the query string

* Fix tests

* Fix frontend for DC with tags (pages / post content)

* Fix colours and typography for DC text toolbar

* Fix DC for content on frontend - 2 and more blocks

* Fix content DC

* Removed 'Date' and 'Modified' relation
Removed 'Get by Author' relation
Removed 'apiFetch' from DC
Removed frontend JS for DC
Refactored DC component logics
Created e2e tests

* Fixed 'author' field
Fixed 'site' type
Fixed async on DC

* Fixed editor side

* Fix date on frontend

* Abstract DC component
Create inspectorTab for DC
Change DateFormatting logics
Create HOC withMaxiDC

* Fixes frontend issue with non custom dates
Moved DC e2e tests folder
Fix tests

* Clean test

* Fix limit value on reset
Fix conflict when having 2 or more DC instances open
Add limit for categories and tax description
Removed 'parent' field for tags
Fix category parent on backend
Fix site settings

* Implement DC for Button Maxi

* Set limit to option labels

* Implement DC for Image Maxi on editor side

* Add front support for Image Maxi DC

* Implement Link support for DC

* Fixes DC links issues

* Fixes users links

* Fix errors
Fix tests

* Fixes tests

* Add async to DC tests

* Skip DC tests

---------

Co-authored-by: moroz-art-dev <moroz.art.dev@gmail.com>
Co-authored-by: Artem <88348091+moroz-art-dev@users.noreply.github.com>
Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>
Co-authored-by: Svitlana <2401618+elzadj@users.noreply.github.com>

elzadj committed:
merge conflicts with master

elzadj committed:
Fix DC for react 18

elzadj committed:
Fix DC blocks rendering

elzadj committed:
Fix dc for old blocks

Marko64 committed:
All blocks

Marko64 committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into Blocks-preview

Naaaaiix committed:
Remove lazy load
Improved withSuspense lifecycle

elzadj committed:
Fix DC default state  on backend

elzadj committed:
Remove comments

Fybex committed:
`getLastChangedBlock`: fix for FSE

Fybex committed:
`entityRecordsWrapper`: fix for FSE `addClearEntity`

Fybex committed:
Small fixes

Fybex committed:
Update package-lock.json

Naaaaiix committed:
Fix tests

Naaaaiix committed:
Remove console.log

Marko64 committed:
Row, pane, slide

elzadj committed:
Fix lint bugs

elzadj committed:
Fix useEffect

elzadj committed:
Try to fix createRoot issue

elzadj committed:
Remove setTimeout for FSE toolbar buttons

Naaaaiix committed:
Implement lighter Suspense

myroslavve committed:
Issue 4394 - calculating margin value for different themes (#4522)

* created `marginValueCalculator`
which calculates `marginValue` as needed and returns precalculated value otherwise

* fix for 2022 and 2023 theme
overwrite important margin styles

* use `MarginValueCalculator` only one time

* fix for first render

* fix imports

* add test to ensure maxi blocks full width is right

* change first letter in function name to lower case

* add styles to the head

* fix error on block in iframe

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into fix-wp-62-beta4-bugs

elzadj committed:
Fix useEffect returning false

elzadj committed:
Merge pull request #4508 from maxi-blocks/fix-wp-62-beta4-bugs

Fix Maxi Blocks for WP 6.2 (beta 5) and 6.1.1

elzadj committed:
Add missing alt tag for parallax images

dependabot[bot] committed:
Bump webpack from 5.72.0 to 5.76.1 (#4548)

Bumps [webpack](https://github.com/webpack/webpack) from 5.72.0 to 5.76.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.72.0...v5.76.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

elzadj committed:
Issue 4510 button (#4539)

* Add aria-label to buttons without text

* Add migrator

* Remove double button for screen reader

* Update button snaps

Naaaaiix committed:
Fixes `withMaxiSuspense` just for the page load

Naaaaiix committed:
Merge branch 'master' into lighter-suspense

Naaaaiix committed:
Fix package-lock.json

elzadj committed:
Deal with version rollback for wp (#4549)

* package

* Clean code and comment out the rollback parts we don't need for now

* Comment out breakpoints setttings

* Revert package-lock.json

elzadj committed:
Fix warning for alt

elzadj committed:
Fix package lock

elzadj committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into issue-4510-parallax-alt

Naaaaiix committed:
Replace the name of the loader HOC

Marko64 committed:
Update maxiBlockComponent.js (#4553)

elzadj committed:
Update src/components/image-alt-control/index.js

Co-authored-by: Josh <50477222+Naaaaiix@users.noreply.github.com>

Naaaaiix committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks

elzadj committed:
Merge pull request #4546 from maxi-blocks/issue-4510-parallax-alt

Add missing alt tag for parallax images

Naaaaiix committed:
Main Maxi block component optimization (#4558)

* Revert Suspense implementation

* Abstract RowBlockTemplate component
Reduce withMaxiProps load
Reduce MaxiBlockComponent load
Abstracted some MaxiBlockComponent methods
Avoid rendering Inspector if it's not selected

* Clean Text Maxi main file

* Clean code

* Library insert optimization (#4559)

* Refactored `onRequestInsertPattern`
Modified Library flow when clicking insert

* Fix package-lock.json

* Fix CORS
Clean code

* Fixes placeholder source

Naaaaiix committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks

Naaaaiix committed:
Merge branch 'master' into lighter-suspense

Naaaaiix committed:
Fix package.json

Naaaaiix committed:
Fix typo

Naaaaiix committed:
Fix tests

Fybex committed:
Merge branch 'master' into issue-4507

Fybex committed:
Merge branch 'lighter-suspense' into issue-4507

Naaaaiix committed:
Fix tests

Fybex committed:
Merge maxiBlockComponent

Naaaaiix committed:
Fixes compatibility with React 18

Fybex committed:
Merge branch 'lighter-suspense' into issue-4507

Fybex committed:
Fix error in FSE

Fybex committed:
Calculate correct `keepStylesOnEditor`

Naaaaiix committed:
New block Loader (#4557)

* Remove lazy load
Improved withSuspense lifecycle

* Fix tests

* Remove console.log

* Implement lighter Suspense

* Fixes `withMaxiSuspense` just for the page load

* Fix package-lock.json

* Replace the name of the loader HOC

* Main Maxi block component optimization (#4558)

* Revert Suspense implementation

* Abstract RowBlockTemplate component
Reduce withMaxiProps load
Reduce MaxiBlockComponent load
Abstracted some MaxiBlockComponent methods
Avoid rendering Inspector if it's not selected

* Clean Text Maxi main file

* Clean code

* Library insert optimization (#4559)

* Refactored `onRequestInsertPattern`
Modified Library flow when clicking insert

* Fix package-lock.json

* Fix CORS
Clean code

* Fixes placeholder source

* Fix package.json

* Fix typo

* Fix tests

* Fix tests

* Fixes compatibility with React 18

Marko64 committed:
Merge branch 'master' of https://github.com/maxi-blocks/maxi-blocks into Blocks-preview

Marko64 committed:
Container and library fix

elzadj committed:
Add preview to Template Library and optimize the path to previews

Marko64 committed:
Alt tags

Fybex committed:
Remove unused imports, clgs

Fybex committed:
Merge branch 'master' into issue-4507

Fybex committed:
Remove commented import

Fybex committed:
Fix FSE preview

Fybex committed:
`getClientIdFromUniqueId` fix for template part

block which is in template(not part)

Fybex committed:
`getLastChangedBlocks `:use undoEdit.name for kind

Marko64 committed:
Row and slide update

Naaaaiix committed:
Fixes #4567

Marko64 committed:
Update edit.js

Marko64 committed:
Update edit.js

Marko64 committed:
Issue 4576 - map empty marker content (#4577)

* Update maxi-map.js

* Update maxi-map.js

Naaaaiix committed:
v1.0.0-RC2 (#4479)

* Upgrade to RC2

* Update tests

elzadj committed:
Fix slide preview

elzadj committed:
Fix pane preview

elzadj committed:
small fix

Fybex committed:
Add `sid`

Fybex committed:
Save and use `sid` instead of `setting`

Fybex committed:
`getTransitionSetting` -> `getIBDataItem`

Fybex committed:
`IBLabelToIDMigrator`

Fybex committed:
Add `sid` support on relations frontend

Fybex committed:
Minor fixes

Fybex committed:
Make possible call`transitionAttributesCreator()`

Fybex committed:
Compare `data-sids` correct

Fybex committed:
Remove obsoleted tests

Fybex committed:
Correct image border `sid`

Naaaaiix committed:
Fixes styles on changing breakpoint

Marko64 committed:
Issue 4482 - List source reset button (#4500)

* Issue 4482 - List source reset button

* Comments

* Update editor.scss

* Removed special class for inputs without reset and CSS for it

elzadj committed:
Fix custom css test for Failed to load resource: net::ERR_FAILED (#4586)

Marko64 committed:
Issue 4581 - SC colour preview square (#4584)

elzadj committed:
Merge pull request #4590 from maxi-blocks/rc2-fixes

Fixes styles on changing breakpoint

Naaaaiix committed:
Adds ContentLoader component to Library

elzadj committed:
Merge pull request #4520 from maxi-blocks/Blocks-preview

Blocks preview

Rustam198804 committed:
Fix styles for toolbar bg color

Marko64 committed:
Issue 4505 - Range tooltip position for long numbers

elzadj committed:
Chahge loading animation from P to H3

Rustam198804 committed:
Fix title for SC library

Fybex committed:
Update e2e

elzadj committed:
Merge pull request #4593 from maxi-blocks/library-loader

Library loader

myroslavve committed:
load default font weight for sc fonts

myroslavve committed:
fix indentation

Rustam198804 committed:
Remove opacity icon

Marko64 committed:
Issue 4514 - DC UI (#4566)

* Issue 4514 - DC UI

1. Placed the icon after the input. The text and the input are one same element, so can't really insert the icon inside of it, only before or after.

3. This one might need some time. Seems like we would need a condition for everything to render a different text, and we will be adding more things, so I say let's skip this for now.

* Styling convention

elzadj committed:
Merge pull request #4595 from maxi-blocks/sc-library-titles

Fixed Title for SC library

Marko64 committed:
Inputs width

Rustam198804 committed:
Fix svg icon for  button icon border

elzadj committed:
Merge pull request #4582 from maxi-blocks/issue-4567

Issue 4567

Marko64 committed:
Update editor.scss

Marko64 committed:
Button label spacing

elzadj committed:
Merge pull request #4589 from maxi-blocks/issue-3790

Issue 3790 & 4222 - IB labels problem

elzadj committed:
Merge pull request #4609 from maxi-blocks/button-icon-border-svg

Fix Svg for  button icon border

elzadj committed:
Merge pull request #4594 from maxi-blocks/issue-4505-range-tooltip-cut-off-big-numbers

Issue 4505 - Range tooltip position for long numbers

elzadj committed:
Merge pull request #4597 from maxi-blocks/issue-4561

Issue 4561 - load default font weight for sc fonts

Fybex committed:
Merge branch 'master' into issue-4507

Rustam198804 committed:
Remove commented line

elzadj committed:
Merge pull request #4598 from maxi-blocks/toolbar-bg-color-ui

Toolbar Background Color UI

elzadj committed:
Merge pull request #4611 from maxi-blocks/issue-4610-minor-button-label-spacing

Issue 4610 minor button label spacing

elzadj committed:
Add 2 more custom css server errors to ignore (#4621)

Rustam198804 committed:
Fix alignment bg color (#4623)

Rustam198804 committed:
Disallowed template library for some blocks (#4617)

Marko64 committed:
Issue 4233 - Button bottom gap remove (#4236)

* Issue 4233 - Button bottom gap remove

* Reversed the logic

* Update maxiStyleCardsTab.js

* tests

* comments

* Update index.js

* Tests

* Update maxi-themes.scss

* Tests

* tests

* Fix `addTypographyStyle`

* tests

* Fix minor bugs
Skip corrupted SC tests

* Skip sc-variable tests

* Map title

* Update style.scss

* Update style.scss

* Update attributes.js

* Tests

* Remove skips

* e2e fix

* Better `showBottomGap` place

---------

Co-authored-by: Josh <jjuancap@gmail.com>
Co-authored-by: Andrew Valenya <46112160+Fybex@users.noreply.github.com>

Fybex committed:
`button-maxi`: fix `blockMigrator` (#4626)

Marko64 committed:
Issue 4322 - Font weight on family change (#4326)

* Font weight on family change

* Delete numberCounterTextMigrator.js

* use closest available font if current is unavailable

* `getClosestAvailableFontWeight` fix

* show warning when selected weight is not in font

* fix inability to reset font family

* fix tests

---------

Co-authored-by: TeksPeks <verstyukm@gmail.com>

elzadj committed:
Merge pull request #4530 from maxi-blocks/issue-4507

Issue 4507 - FSE. IB work fix on pattern inserting/block duplication/copy-paste

Rustam198804 committed:
Merge branch 'master' into issue-3788

Rustam198804 committed:
Fix some small bugs
